demos/instr.h

```diff
@@ -77,4 +77,58 @@ inline void JumpToAfterSpeculation() {
   asm volatile("b afterspeculation");
 }
 #endif
+
+#ifdef __i386__
+// Returns the original value of FS and sets the new value.
+__attribute__((always_inline))
```

Comment:
All of these new functions are marked `__attribute__((always_inline))`. How many of them really need to be inlined in order to work?

Reply:
CPUID barrier must be inlined, so that Intel does not jump over it Spectre v2-style. The fetchers are useful to be inlined, because they are executed in the speculation window - not inlining them would decrease reliability. The setters do not have to be inlined.

Reply:
Ack, curse the branch predictor. But I guess that's why we're here in the first place. :)

This explanation will be valuable in a comment so some hapless mattdr doesn't come along later and think it can be removed.

Reply:
Done.